### PR TITLE
Replicate apply method in the DynamicClient

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Replicate apply method in the DynamicClient.

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: plugins/module_utils/common.py [hunk 1]
```diff
@@ -27,19 +27,20 @@
 from distutils.version import LooseVersion
 
 from ansible_collections.kubernetes.core.plugins.module_utils.args_common import (AUTH_ARG_MAP, AUTH_ARG_SPEC)
+from ansible_collections.kubernetes.core.plugins.module_utils.apply import apply, apply_object
+from ansible_collections.kubernetes.core.plugins.module_utils.exceptions import ApplyException
+from ansible_collections.kubernetes.core.plugins.module_utils.k8sdynamicclient import K8SDynamicClient
```
Comment (goneri) [inline plugins/module_utils/common.py:32]: You need to remove this line since you doing the same after.

File: plugins/module_utils/apply.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
+# Copyright [yyyy] [name of copyright owner]
```
Comment (Akasurde) [inline plugins/module_utils/apply.py:1]:
```suggestion
# Copyright [2017] [Red Hat, Inc.]
```

File: plugins/module_utils/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Copyright [yyyy] [name of copyright owner]
```
Comment (Akasurde) [inline plugins/module_utils/exceptions.py:1]:
```suggestion
# Copyright [2017] [Red Hat, Inc.]
```

File: plugins/module_utils/k8sdynamicclient.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Copyright [yyyy] [name of copyright owner]
```
Comment (Akasurde) [inline plugins/module_utils/k8sdynamicclient.py:1]:
```suggestion
# Copyright [2017] [Red Hat, Inc.]
```

File: tests/unit/module_utils/test_apply.py [hunk 5]
```diff
@@ -0,0 +1,421 @@
+# Copyright [yyyy] [name of copyright owner]
```
Comment (Akasurde) [inline tests/unit/module_utils/test_apply.py:1]:
```suggestion
# Copyright [2017] [Red Hat, Inc.]
```

File: plugins/module_utils/apply.py (hunk 2 above)
Comment (goneri) [reply plugins/module_utils/apply.py]: why 2017? it should be 2021.

**Review by gravesm — APPROVED** [review]
Great work on this!

**Review by Akasurde — APPROVED** [review]

**Pull request merged**